hawc/apps/epiv2/models.py

```diff
@@ -0,0 +1,322 @@
+    percentile25 = models.FloatField(blank=True, null=True,)
+    percentile75 = models.FloatField(blank=True, null=True,)
+    maximum = models.FloatField(blank=True, null=True,)
+    neg_exposure = models.FloatField(
+        verbose_name="Percent with negligible exposure",
+        help_text="e.g., %% below the LOD",
+        blank=True,
+        null=True,
+    )
+    comments = models.CharField(
+        max_length=128, verbose_name="Exposure level comments", blank=True, null=True,
+    )
+    data_location = models.CharField(
+        max_length=128, help_text="e.g., table number", blank=True, null=True,
+    )
+    created = models.DateTimeField(auto_now_add=True)
+    last_updated = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        return self.name
+
+
+class Outcome(models.Model):
+    objects = managers.OutcomeManager()
+
+    name = models.CharField(
+        max_length=64,
+        help_text="A unique name for this health outcome that will help you identify it later.",
+    )
+    study_population = models.ForeignKey(
+        StudyPopulation, on_delete=models.CASCADE, related_name="outcomes"
+    )
+    health_outcome = models.CharField(max_length=128,)
+    health_outcome_system = models.CharField(
+        max_length=128,
+        choices=constants.HealthOutcomeSystem.choices,
+        help_text="If multiple cancer types are present, report all types under Cancer.",
+        blank=True,
+        null=True,
+    )
+    created = models.DateTimeField(auto_now_add=True)
+    last_updated = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        return self.name
+
+
+class AdjustmentFactor(models.Model):
+    objects = managers.AdjustmentFactorManager()
+
+    name = models.CharField(
+        max_length=64,
+        help_text="A unique name for this adjustment factor that will help you identify it later.",
+    )
+    study_population = models.ForeignKey(
+        StudyPopulation, on_delete=models.CASCADE, related_name="adjustment_factors"
+    )
+    description = models.CharField(max_length=128, blank=True, null=True,)
+    created = models.DateTimeField(auto_now_add=True)
+    last_updated = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        return self.name
+
+
+class DataExtraction(models.Model):
+    objects = managers.DataExtractionManager()
+
+    study_population = models.ForeignKey(
+        StudyPopulation, on_delete=models.CASCADE, related_name="data_extractions"
+    )
+    sub_population = models.CharField(max_length=64,)
+    outcome = models.ForeignKey(Outcome, on_delete=models.CASCADE, blank=True, null=True,)
+    exposure_level = models.ForeignKey(
+        ExposureLevel, on_delete=models.SET_NULL, blank=True, null=True
+    )
+    measurement_timing = models.CharField(max_length=128, blank=True, null=True,)
+    n = models.PositiveIntegerField(blank=True, null=True,)
+    effect_estimate_type = models.CharField(
```

Comment:
For CharField and TextFIeld, never set null=True; blank=True is fine, b/c otherwise you'll have empty strings and nulls in the database and that makes typing really complicated